src/transaction.js

```diff
@@ -242,6 +252,14 @@ Transaction.prototype.getId = function () {
   return bufferutils.reverse(this.getHash()).toString('hex')
 }
 
+Transaction.prototype.getNormalizedHash = function () {
```

Comment:
I honestly think this function can be achieved much cleaner without using and modifying `hashForSignature`:

```js
function getNormalizedHash = function(tx) {
  var txTmp = tx.clone()

  txTmp.ins.forEach(function(txIn) {
    txIn.script = Script.EMPTY
  })

  var hashTypeBuffer = new Buffer(4)
  hashTypeBuffer.writeInt32LE(Transaction.SIGHASH_ALL, 0)

  return crypto.hash256(Buffer.concat([txTmp.toBuffer(), hashTypeBuffer]))
}
```

@dcousens put up a very nice summary of the whole debate. IMHO, aligning bitcoinjs' features with bitcoin core as much as possible provides consistency and least surprise for devs. I'm not comfortable merging a feature explicitly rejected by the core team after much debate, especially when this feature is very easy to implement when one indeed needs it for their own purposes.